### PR TITLE
SB-754 supporting JSON in the REST

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

File: strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/configuration/ProxyConfiguration.java [hunk 1]
```diff
@@ -39,8 +39,7 @@
      */
     @XmlAttribute
     private String type;
-
-    @XmlElement(name = "host")
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/configuration/ProxyConfiguration.java]: Not sure why this was removed...? (I thought that `non-proxy-host` was too long, so I thought of just using `host` instead. Is this causing a problem?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseArtifactController.java [hunk 2]
```diff
@@ -26,32 +24,6 @@
     // ----------------------------------------------------------------------------------------------------------------
     // Common-purpose methods
 
-    protected synchronized <T> T read(String json,
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseArtifactController.java]: Thanks for killing this monster! :D

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapGroupSearchResponseEntityBody.java [hunk 3]
```diff
@@ -19,13 +19,13 @@
...
+    LdapGroupSearchResponseEntityBody searchBase(String searchBase)
     {
         this.searchBase = searchBase;
         return this;
     }
 
-    LdapGroupSearchXmlHolder searchFilter(String searchFilter)
+    LdapGroupSearchResponseEntityBody searchFilter(String searchFilter)
     {
         this.searchFilter = searchFilter;
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapGroupSearchResponseEntityBody.java]: Ignore.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapUserDnPatternsResponseEntityBody.java [hunk 4]
```diff
@@ -11,17 +11,17 @@
...
 
-    LdapUserDnPatternsXmlHolder()
+    LdapUserDnPatternsResponseEntityBody()
     {
     }
 
-    public LdapUserDnPatternsXmlHolder(List<String> userDnPatterns)
+    public LdapUserDnPatternsResponseEntityBody(List<String> userDnPatterns)
     {
         this.userDnPatterns = userDnPatterns;
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapUserDnPatternsResponseEntityBody.java]: Ignore.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapUserSearchResponseEntityBody.java [hunk 5]
```diff
@@ -19,13 +19,13 @@
     @XmlElement(name = "searchFilter")
     private String searchFilter;
 
-    LdapUserSearchXmlHolder searchBase(String searchBase)
+    LdapUserSearchResponseEntityBody searchBase(String searchBase)
     {
         this.searchBase = searchBase;
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapUserSearchResponseEntityBody.java]: Ignore.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapUserSearchResponseEntityBody.java [hunk 6]
```diff
@@ -19,13 +19,13 @@
...
         return this;
     }
 
-    LdapUserSearchXmlHolder searchFilter(String searchFilter)
+    LdapUserSearchResponseEntityBody searchFilter(String searchFilter)
     {
         this.searchFilter = searchFilter;
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/LdapUserSearchResponseEntityBody.java]: Ignore.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/support/ErrorResponseEntityBody.java [hunk 7]
```diff
@@ -10,17 +10,17 @@
...
 
-    public ErrorXmlHandler()
+    public ErrorResponseEntityBody()
     {
     }
 
-    public ErrorXmlHandler(String message)
+    public ErrorResponseEntityBody(String message)
     {
         this.message = message;
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/support/ErrorResponseEntityBody.java]: Ignore.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/ConfigurationManagementControllerTest.java [hunk 8]
```diff
@@ -118,15 +113,10 @@ public void testSetAndGetGlobalProxyConfiguration()
 
         logger.debug("Current proxy host: " + proxyConfiguration.getHost());
 
-        String response = given().contentType(MediaType.APPLICATION_XML_VALUE)
+        ProxyConfiguration pc = given().contentType(MediaType.APPLICATION_XML_VALUE)
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/ConfigurationManagementControllerTest.java]: Cosmetic: Looks good, could you just fix the indentation, please? :)

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/ConfigurationManagementControllerTest.java [hunk 9]
```diff
@@ -195,28 +185,17 @@ public void testAddGetStorage()
     }
 
     private Storage getStorage(String storageId)
-            throws IOException, JAXBException
     {
 
         String url = getContextBaseUrl() + "/configuration/strongbox/storages/" + storageId;
 
-        String response = given().contentType(MediaType.TEXT_PLAIN_VALUE)
+        return given().contentType(MediaType.TEXT_PLAIN_VALUE)
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/ConfigurationManagementControllerTest.java]: Cosmetic: Looks good, could you just fix the indentation, please? :)

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/UserControllerTest.java [hunk 10]
```diff
@@ -195,20 +181,17 @@ public void testGenerateSecurityToken()
...
-                                 .extract()
-                                 .asString();
-        user = objectMapper.readValue(response, User.class);
+                                 .as(User.class);
+
         assertNotNull(user.getSecurityTokenKey());
 
         //3. Generate token
-        response = given().when()
+        String response = given().when()
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/UserControllerTest.java]: Cosmetic: Looks good, could you just fix the indentation, please? :)

**Comment by carlspring** [comment]
Thanks for sorting this out! It looks like I had implemented it before when we were still using Jersey (with some improvements, of course ;) ).

File: strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/configuration/ProxyConfiguration.java (hunk 1 above)
Comment (fuss86) [inline strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/configuration/ProxyConfiguration.java]: Yes. There was an error related with ambiguous host xml element. See above in this file, there already exists

> @XmlAttribute
    private String host;

**Comment by fuss86** [comment]
@carlspring it's ready to be merged

**Pull request merged**